doc/rtt.js

```diff
@@ -0,0 +1,178 @@
+// TODO: Become able to run JSdoc in .h/.cpp files, with a JSdoc plugin a la https://www.npmjs.com/package/jsdoc-jsx
+
+/**
+ *
+ * The <tt>pc-nrfjprog-js.RTT</tt> module exposes the functionality to the RTT functionality.
+ *
+ * @example
+ * const nrfjprogjs = require('pc-nrfjprog-js');
+ * const RTT = nrfjprogjs.RTT;
+ *
+ * RTT.start(12345678, {}, function(err, down, up) {
+ *      RTT.write(0, 'Some data to write', function(err, length, timeSinceRTTStartInUs) {
+ *          RTT.read(0, 100, function(err, stringData, rawData, timeSinceRTTStartInUs) {
+ *              RTT.stop(function(err) {
+ *                  console.log('Stopped');
+ *              });
+ *          });
+ *      });
+ * });
+ *
+ * @module pc-nrfjprog-js.RTT
+ */
+
+/**
+ * Possible error.<br/>
+ * If an operation completed sucessfully, the error passed to the callback
+ * function will be <tt>undefined</tt> (and thus, falsy).<br/>
+ * This will be an instance of the built-in {@link https://nodejs.org/dist/latest/docs/api/errors.html#errors_class_error|Error} class, with some extra properties:
+ * @typedef {Error} Error
+ * @property {integer} errno
+ *    The error number. Value will be one of the following predefined constants:<br/>
+ *    <tt>nrfjprogjs.RTTCouldNotLoadHighlevelLibrary</tt><br/>
+ *    <tt>nrfjprogjs.RTTCouldNotOpenHighlevelLibrary</tt><br/>
+ *    <tt>nrfjprogjs.RTTCouldNotGetDeviceInformation</tt><br/>
+ *    <tt>nrfjprogjs.RTTCouldNotLoadnRFjprogLibrary</tt><br/>
+ *    <tt>nrfjprogjs.RTTCouldNotOpennRFjprogLibrary</tt><br/>
+ *    <tt>nrfjprogjs.RTTCouldNotConnectToDevice</tt><br/>
+ *    <tt>nrfjprogjs.RTTCouldNotStartRTT</tt><br/>
+ *    <tt>nrfjprogjs.RTTCouldNotFindControlBlock</tt><br/>
+ *    <tt>nrfjprogjs.RTTCouldNotGetChannelInformation</tt><br/>
+ *    <tt>nrfjprogjs.RTTCouldNotCallFunction</tt><br/>
+ *    <tt>nrfjprogjs.RTTNotInitialized</tt><br/>
+ *    <tt>nrfjprogjs.RTTCouldNotExecuteDueToLoad</tt><br/>
+ *
+ * @property {String} errcode A human-readable version of the error code.
+ * @property {String} erroroperation The internal function that caused the error.
+ * @property {String} errmsg Error string. The value will be equal to that of the built-in <tt>message</tt> property.
+ * @property {integer} lowlevelErrorNo The low-level error code, if applicable.
+ * @property {String} lowlevelError A human-readable version of the low-level error code.
+ * @property {String} log The complete log from the internal functions.
+ *
+ * @example
+ * RTT0.start(12345678, {}, function(err, down, up){
+ *     if (err) {
+ *         throw err;
+ *     } else {
+ *         // success
+ *     }
+ * });
+ *
+ * @example
+ * RTT.start(12345678, {}, function(err, down, up){
+ *     if (err && err.errno === nrfjprogjs.RTTCouldNotFindControlBlock) {
+ *          console.error('The firmware is not RTT capable');
+ *     }
+ * });
+ */
+
+/**
+ * Information about the different up and down channels available on the device
+ * @typedef ChannelInfo
```

Comment:
Can a user tell if a `ChannelInfo` is for a downstream or upstream channel?

Reply:
At the moment you can only separate between them based on which parameter they are in the start callback. But it would be simple to add in the `ChannelInfo` type as well.